design-system/react/src/components/ButtonLink/ButtonLink.tsx

```diff
@@ -1,28 +1,37 @@
-import { cx } from '@fuel-ui/css';
+import type { LayerVariant } from '@fuel-ui/css';
+import { createElement } from 'react';
+import { mergeProps } from 'react-aria';
 
-import { createComponent } from '../../utils';
-import type { HTMLProps } from '../../utils';
+import { _unstable_createComponent } from '../../utils';
 import type { ButtonProps } from '../Button';
 import { Button } from '../Button';
 
-export type ButtonLinkProps = ButtonProps &
-  HTMLProps['a'] & {
-    isExternal?: boolean;
-  };
+import type * as t from './defs';
 
-export const ButtonLink = createComponent<ButtonLinkProps>(
-  ({ as = 'a', role = 'link', isExternal, className, ...props }) => {
-    const customProps = {
-      ...props,
-      className: cx('fuel_ButtonLink', className),
+import { Components } from '~/defs';
+import { useStyles, useElementProps, createStyle } from '~/hooks';
+
+export const ButtonLink = _unstable_createComponent<t.ButtonLinkDef>(
+  Components.ButtonLink,
+  ({ isExternal = false, as = 'a', role = 'link', size, ...props }) => {
```

Comment:
since it's undefined it's already a falsy value, you don't need this false here

```suggestion
  ({ isExternal, as = 'a', role = 'link', size, ...props }) => {
```